xtask/src/commands/api_extractor.rs

```diff
@@ -0,0 +1,36 @@
+use anyhow::{Context, Result};
+use clap::{Parser, Subcommand};
+
+use crate::utils::command::run_command;
+
+#[derive(Debug, Parser)]
+pub struct ApiExtractorCmd {
+  #[clap(subcommand)]
+  sub_cmd: ApiExtractorSubCmd,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum ApiExtractorSubCmd {
+  Update,
+  Ci,
+}
+
+impl ApiExtractorCmd {
+  pub fn run(&self) -> Result<()> {
+    match &self.sub_cmd {
+      ApiExtractorSubCmd::Update => {
+        run_command("pnpm", &["-w", "build:js"])?;
+        run_command(
+          "pnpm",
+          &["--filter", "@rspack/*", "api-extractor", "--local"],
+        )?;
+      }
+      ApiExtractorSubCmd::Ci => {
+        run_command("pnpm", &["--filter", "@rspack/*", "api-extractor:ci"]).with_context(|| {
+          "Api-extractor testing failed. Did you forget to update the snapshots locally?\nRun the command below locally to fix this error (in the *ROOT* of rspack workspace).\n$ ./x api-extractor update"
+        })?;
+      }
+    }
```

Comment:
what's the benefit of run pnpm script in xtask then in justfile directly

Reply:
You can use cmd args more flexibly. This is just a start, and perhaps `api-extractor` isn't the best choice. I expect to implement commands like `debug`, `publish`, and `version` in `xtask`.

Reply:
I'm not sure put js related task into xtask is a good idea, maybe we should start with rust related task first, @h-a-n-a what's your idea?